### PR TITLE
Add CodeQL static analysis

### PR DESCRIPTION
**Pull request by emabrey**
Add workflow that runs static analysis every 6 hours daily.

Signed-off-by: Emily Mabrey <emabrey@tenacityaudio.org>

This should help make it easier to find and fix any C++ or Python security issues. This uses CodeQL, an open-source static analyzer available here: https://securitylab.github.com/tools/codeql/.

<details>
<summary>Checklist</summary>

- [x] I have signed off my commits using `-s` or `Signed-off-by`\* (See: [Contributing § DCO](https://github.com/tenacityteam/tenacity/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))
- [x] I made sure the code compiles on my machine
- [x] I made sure there are no unnecessary changes in the code\*
- [x] I made sure the title of the PR reflects the core meaning of the issue you are solving\*
- [x] I made sure the commit message(s) contain a description and answer the question "Why do those changes fix that particular issue?" or "Why are those changes really necessary as improvements?"\*

\* indicates required

</details>

### CODE REVIEW

**Comment by Semisol** [comment]
`Error: We were unable to automatically build your code. Please replace the call to the autobuild action with your custom build steps.  Failure invoking /opt/hostedtoolcache/CodeQL/0.0.0-20210702/x64/codeql/cpp/tools/autobuild.sh with arguments .`
Also you might want to run it for every commit, especially for PRs.

File: .github/workflows/codeql-analysis.yml [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+      run: |
+        exec bash "scripts/ci/configure.sh"
+    - name: Build
+      run: |
+        exec bash "scripts/ci/build.sh"
+    - name: "Print compiler cache stats"
+      run: ${{ matrix.config.compiler_cache }} -s
+    - name: Install
+      run: |
+        exec bash "scripts/ci/install.sh"
```
Comment (Semisol) [inline .github/workflows/codeql-analysis.yml:112]: You removed the analysis part.
```suggestion
        exec bash "scripts/ci/install.sh"
    - name: Perform CodeQL Analysis
      uses: github/codeql-action/analyze@v1
```

**Review by Semisol — CHANGES_REQUESTED** [review]

**Comment by nbsp** [comment]
I don't like the idea of using GitHub for this. This (a) increases our dependence on GitHub, and (b) wouldn't work on patches sent by email.

**Comment by emabrey** [comment]
@SFR-git If we transition from Github it just stops working. It doesn't add any extra dependence on Github that we don't already have. I could just add this to the `cmake_build.yml` so it's more like modifying what is already there. The only reason it's in a separate file is for cron job support.

**Comment by nbsp** [comment]
The issue with patches still stands. Could you edit the builds.sr.ht manifest to use this as well?

**Comment by emabrey** [comment]
> 
> 
> The issue with patches still stands. Could you edit the builds.sr.ht manifest to use this as well?

No. It would violate the license. https://securitylab.github.com/tools/codeql/license/

I view this as an extra tool for GitHub usage only, not as the sole vulnerability scanning allowed. We can always implement an additional GitHub independent vulnerability scanner. The only reason I am using this one is because it seamlessly integrates with the GitHub "Security" tab. If we leave GitHub then we don't have a Security tab, so we would just stop using it by deleting the workflow at that time. This doesn't have to be our only vulnerability scanner; I view this as an addendum not an exclusive. I also do not plan to make this a required check for merges or patches. This is literally just an information populating thing, not a build CI thing.

**Comment by aeiouaeiouaeiouaeiouaeiouaeiou** [comment]
Sorry to bother but does the CodeQL analysis trigger too often? I don't want to change it for myself as I maintaining the mirror.

**Comment by Semisol** [comment]
Wdym, it triggers on pushes but does not constantly spam us.

**Pull request merged**